### PR TITLE
add docs

### PR DESCRIPTION
**Pull request by iscai-msft**

### CODE REVIEW

**Comment by pakrym** [comment]
@chamons @ShivangiReja PTAL and merge.

File: docs/client/initializing.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+You can use your client if you import the namespace you specified when generating (under flag `--namespace`). For the sake of this example,
+let's say the namespace is `Azure.Pets` and your client is `PetsClient`. You could then access the client if you import the namespace.
+
+```csharp
+using Azure.Pets;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+string endpoint = "http://localhost:3000";
```
Comment (chamons) [inline docs/client/initializing.md:16]: nit: `const`?

File: docs/client/operations.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
+# <img align="center" src="../images/logo.png">  Calling Operations with Your C# Client
+
+AutoRest provides both synchronous and asynchronous method overloads for each service operation.
+Depending on your swagger definition, operations can be accessed through operation groups (TODO: link to swagger docs) on the client,
```
Comment (chamons) [inline docs/client/operations.md:4]: Does this TODO need an issue for tracking or was it expected to be handled in this PR?

File: docs/client/operations.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+```csharp
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Pets;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+var client = new PetsClient(endpoint);
+Dog dog = (await client.GetDogAsync()).Value;
```
Comment (chamons) [inline docs/client/operations.md:47]: Isn't there an implicit Operation<Dog> -> Dog conversion that does the .Value for us here?

File: docs/client/operations.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+```
+
+## Long Running Operations
+
+Long-running operations are operations which consist of an initial request sent to the service to start an operation, followed by polling the service at intervals to determine whether the operation has completed or failed, and if it has succeeded, to get the result.
+
+In concurrence with our [.NET guidelines][poller_guidelines], all of our long running operations are prefixed with `Start`, to signify the starting of the long running operation.
```
Comment (chamons) [inline docs/client/operations.md:54]: nit phrasing. You call if you take it.
```suggestion
In concurrence with our [.NET guidelines][poller_guidelines], all long running operations are prefixed with `Start`, to signify the starting of a long running operation.
```

File: docs/client/operations.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+using Azure.Core.Pipeline;
+
+var client = new PollingPagingExampleClient(endpoint);
+
+var product = new Product
+{
+    Id = 1,
+    Name = "My Polling Example"
+};
+var outputProduct = (await client.StartBasicPolling(product).WaitForCompletionAsync()).Value;
```
Comment (chamons) [inline docs/client/operations.md:109]: Same question on Operation<T>.

**Review by chamons — APPROVED** [review]
A few nit picks, but ❤️ the work. Docs are awesome, and very much needed. :)

**Comment by chamons** [comment]
Hey @iscai-msft - I tweaks the text. Let me know if you have any thoughts/concerns. 

I'll land it early next week otherwise. 

@ShivangiReja  - Thoughts?

**Comment by iscai-msft** [comment]
oh thanks so much @chamons I kept forgetting to address your comments, that's totally my b

**Comment by ShivangiReja** [comment]
@chamons what do you think if we also add c# flag definitions somewhere in the docs? I see we already have few flag definitions [here](https://github.com/Azure/autorest/blob/master/docs/generate/flags.md#net-flags) but do you think we should add these in this repo, so in future if we add new c# flag, we will remember to update the definition in the docs?

Few flag definitions are missing in [this ](https://github.com/Azure/autorest/blob/master/docs/generate/flags.md#net-flags) doc like CredentialTypes, CredentialScopes, CredentialHeaderName..etc.

**Comment by iscai-msft** [comment]
hey @ShivangiReja I think @lmazuel wants all of the flags in one place. There's a section for .net specific, and also for general. In the general section I have credential scopes and I can add the others. Can you lmk which flags I missed?

File: docs/generate/readme.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# <img align="center" src="../images/logo.png">  Generating C# Clients with AutoRest
+
+Most of the information you'll need to generate a Python client can be found in the general docs [here][general]. In these docs, we go over a couple C#-specific scenarios.
```
Comment (ShivangiReja) [inline docs/generate/readme.md:3]: nit: python -> c#

**Comment by ShivangiReja** [comment]
> hey @ShivangiReja I think @lmazuel wants all of the flags in one place. There's a section for .net specific, and also for general. In the general section I have credential scopes and I can add the others. Can you lmk which flags I missed?

What do you think of adding [this](https://github.com/Azure/autorest/blob/master/docs/generate/flags.md#net-flags) link somewhere in the docs? 

[These](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/AutoRest/Plugins/Configuration.cs#L68-L75) are the newly added flags in c#. Last 3 flags are management related.

File: docs/faq.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# <img align="center" src="./images/logo.png">  FAQ
+
+1. What version of AutoRest C# should I use?
+    If your project is a part of `azure-sdk-for-net` repo please follow the steps in [Using C# generation in azure-sdk-net repo][generate_with_sdk_repo] and avoid using autorest command line directly.
```
Comment (ShivangiReja) [inline docs/faq.md:4]: nit: new line :) 
https://github.com/Azure/autorest.csharp/blob/2c9fcd46eac603d9154cc36cf63a0d33bccf0180/docs/faq.md

File: docs/faq.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# <img align="center" src="./images/logo.png">  FAQ
+
+1. What version of AutoRest C# should I use?
```
Comment (ShivangiReja) [inline docs/faq.md:3]: Should we also add this question?

>How do I generate code based on V2 version?
   You can add --legacy flag to the command line to get the previous behavior.

**Review by ShivangiReja — APPROVED** [review]
@iscai-msft thanks for adding these docs :)

**Comment by chamons** [comment]
@ShivangiReja  @iscai-msft  - Doc changes are up

**Comment by iscai-msft** [comment]
@chamons can't approve bc it's my own pr, but wanted to thank you for your work and say it looks great to me!

**Review by chamons — APPROVED** [review]
Approving again on behalf of @iscai-msft

**Review by ShivangiReja — APPROVED** [review]

**Pull request merged**